src/lib.rs

```diff
@@ -165,18 +166,19 @@ pub fn run(config: &Config) -> Vec<ViolationWithContext> {
 }
 
 #[allow(clippy::too_many_arguments)]
-fn run_fixing_loop(
+fn run_fixing_loop<'a>(
     violations: &mut Vec<ViolationWithContext>,
-    file_contents: &mut Vec<u8>,
+    file_contents: impl Into<MutRopeOrSlice<'a>>,
```

Comment:
To accommodate the `tree-sitter-lint-lsp` use case of having its file contents as a `Rope`, exposed a `MutRopeOrSlice` enum (similar to how `tree_sitter_grep` is exposing an (immutable) `RopeOrSlice` enum)